srcpkgs/libopensmtpd/template

```diff
@@ -0,0 +1,31 @@
+# Template file for 'libopensmtpd'
+pkgname=libopensmtpd
+version=0.7
+revision=1
+hostmakedepends="mdocml"
+makedepends="libevent-devel"
+short_desc="Library for creating opensmtpd filters"
+maintainer="Adam Gausmann <agausmann@fastmail.com>"
+license="custom:BSD-like"
+homepage="http://imperialat.at/dev/libopensmtpd/"
+distfiles="https://distfiles.sigtrap.nl/${pkgname}-${version}.tar.gz"
+checksum=cdd6d5e11a7063f9265da43a42309c646554b8b68b03982ca8472dbe2b08a613
+
+libopensmtpd-devel_package() {
+	short_desc+=" - development files"
+	depends="${sourcepkg}>=${version}_${revision}"
+	pkg_install() {
+		vmove usr/include
+		vmove "usr/lib/*.so"
+		vmove usr/share/man/man3
+	}
+}
+
+do_build() {
+	make -f Makefile.gnu
```

Comment:
`makecmd="make -f Makefile.gnu"` instead of the custom `do_build` and `do_install` functions should work.

Reply:
I've tried a gnu-makefile build style just like you suggested (also making sure move the `vlicense LICENSE` to `post_install`), but I get this error:
```
opensmtpd.c:19:10: fatal error: sys/tree.h: No such file or directory
   19 | #include <sys/tree.h>
      |          ^~~~~~~~~~~~
compilation terminated.
```
Adding `libbsd-devel` to makedepends doesn't seem to fix it either.

Reply:
It is missing some cflags by the look of it:

- with manual `do_build`:
```
cc -fstack-clash-protection -D_FORTIFY_SOURCE=2 -mtune=generic -O2 -pipe    -fdebug-prefix-map=/builddir/libopensmtpd-0.7=. -I/builddir/libopensmtpd-0.7/ -I/builddir/libopensmtpd-0.7//openbsd-compat/ -Wall -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -Wshadow -Wpointer-arith -Wcast-qual -Wsign-compare -fPIC -DNEED_RECALLOCARRAY=1 -DNEED_STRLCAT=1 -DNEED_STRLCPY=1 -DNEED_STRTONUM=1      -c -o opensmtpd.o opensmtpd.c
```

- with `build_style=gnu-makefile; make_cmd="make -f Makefile.gnu"`:
```
cc -fstack-clash-protection -D_FORTIFY_SOURCE=2 -mtune=generic -O2 -pipe    -fdebug-prefix-map=/builddir/libopensmtpd-0.7=.      -c -o opensmtpd.o opensmtpd.c
```

Reply:
Try `make_use_env=yes`.

Reply:
Much better than the patch file I was thinking of, thanks!